File: php/zmq/bench/parallel.php
<?php

$pid   = 1;
$count = $argv[1];
$cmd   = isset($argv[2]) ? 1 : 0;
$max   = 10000;
$loop  = $max / $count;

while($pid && $count--){
    $pid = pcntl_fork();
    if($pid == -1) exit('cant fork');
}

$status = null;
if( $pid ){
    error_log('PARENT: waiting...');
    //pcntl_wait($status);
    //pcntl_waitpid(-1, $status, WNOHANG);
    pcntl_waitpid(-1, $status);
}
else{
    error_log("CHILD : loop=$loop, ". getmypid());
    if($cmd){
        exec("php memcached_client.php $loop");
    }
    else {
        exec("php client.php 0 $loop");
    }
}

File: php/zmq/bench/parallel.pl
#!/usr/bin/perl

use strict;
use warnings;
use Getopt::Long;

GetOptions('-n=s' => \$::opts{n});
die if(!defined $::opts{n});

my $pid   = 1;
my $count = $::opts{n};
my $max   = 10000;
my $loop  = $max / $count;

while($pid && $count--){
    $pid = fork;
    die "Cannot fork: $!" unless defined $pid;
}

if( $pid ){
    # 子プロセスの終了を待機する。
    warn "親プロセス[$$]: waiting...";
    wait;
}
else{
    # 子プロセスで2秒待つ
    warn "子プロセス[$$]: 処理開始";
    #sleep 1;
    #system("php client.php 0 $loop")
    system("php memcached_client.php $loop")
}
